### PR TITLE
Fix #16997 - assertion of first/classic theme from unsorted list

### LINKED ISSUE #16997: Unit tests from FolderThemeCatalogTest class are failing
**Describe the bug**
Unit tests from `Tests\Unit\Core\MailTemplate\FolderThemeCatalogTest` class (`tests/Unit/Core/MailTemplate/FolderThemeCatalogTest.php` file) are failing.

The issue seems to be related with the new `modern` theme because the tests are written to expect the `classic` theme by default.

For unknown reasons the tests are not failing on Travis CI, but are failing on other enviroments like GitLab CI.

**To Reproduce**
1. run tests in GitlabCI and/or under different OS/filesystem

**Screenshots**
```
$ bash tests/check_unit.sh
> @php -d date.timezone=UTC ./vendor/bin/phpunit -c tests/Unit/phpunit.xml
PHPUnit 5.7.27 by Sebastian Bergmann and contributors.

...............................................................  63 / 785 (  8%)
............................................................... 126 / 785 ( 16%)
............................................................... 189 / 785 ( 24%)
............................................................... 252 / 785 ( 32%)
............................................................... 315 / 785 ( 40%)
............................................................... 378 / 785 ( 48%)
............................................................... 441 / 785 ( 56%)
............................................................... 504 / 785 ( 64%)
............................................................... 567 / 785 ( 72%)
...............................F.F.FF.......................... 630 / 785 ( 80%)
............................................................... 693 / 785 ( 88%)
............................................................... 756 / 785 ( 96%)
.............................                                   785 / 785 (100%)

Time: 3.02 seconds, Memory: 40.00MB

There were 4 failures:

1) Tests\Unit\Core\MailTemplate\FolderThemeCatalogTest::testListThemes
Expectation failed for method name is equal to <string:dispatchWithParameters> when invoked at sequence index 0
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'classic'
+'modern'

/builds/mvorisek/prestashop_eshop/src/Core/MailTemplate/FolderThemeCatalog.php:93
/builds/mvorisek/prestashop_eshop/tests/Unit/Core/MailTemplate/FolderThemeCatalogTest.php:99

2) Tests\Unit\Core\MailTemplate\FolderThemeCatalogTest::testInvalidTheme
Failed asserting that exception message 'Invalid requested theme "unknown", only available themes are: modern, classic' contains 'Invalid requested theme "unknown", only available themes are: classic, modern'.

3) Tests\Unit\Core\MailTemplate\FolderThemeCatalogTest::testListThemesWithoutCoreFolder
Expectation failed for method name is equal to <string:dispatchWithParameters> when invoked at sequence index 0
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'classic'
+'modern'

/builds/mvorisek/prestashop_eshop/src/Core/MailTemplate/FolderThemeCatalog.php:93
/builds/mvorisek/prestashop_eshop/tests/Unit/Core/MailTemplate/FolderThemeCatalogTest.php:215

4) Tests\Unit\Core\MailTemplate\FolderThemeCatalogTest::testListThemesWithoutModulesFolder
Expectation failed for method name is equal to <string:dispatchWithParameters> when invoked at sequence index 0
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'classic'
+'modern'

/builds/mvorisek/prestashop_eshop/src/Core/MailTemplate/FolderThemeCatalog.php:93
/builds/mvorisek/prestashop_eshop/tests/Unit/Core/MailTemplate/FolderThemeCatalogTest.php:228

FAILURES!
Tests: 785, Assertions: 2033, Failures: 4.
Script @php -d date.timezone=UTC ./vendor/bin/phpunit -c tests/Unit/phpunit.xml handling the unit-tests event returned with error code 1
UNIT TESTS FAILED
```

**Additional information**
PrestaShop version: develop and/or 1.7.6.1
PHP version: 7.2

Can you please verify and advise/fix this issue? As it breaks tests in other than non-Travis enviroments it should be fixed asap. Thanks.

### PR DESCRIPTION
**Pull request by mvorisek**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Fixes #16997 - Unit tests from FolderThemeCatalogTest class are failing
| Type?         | bug fix
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #16997
| How to test?  | All tests need to pass.

Issue introduced at 1.7.6.x as 1.7.5.x does not contain modern/multiple mail themes. Please merge into 1.7.6.x and develop asap. Thanks.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16998)

### CODE REVIEW

**Review by Progi1984 — APPROVED** [review]
Need a second review

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @mvorisek

**Pull request merged**
